src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -99,10 +110,5 @@ type MessageDbReader internal (connectionString : string, leaderConnectionString
         use! reader = cmd.ExecuteReaderAsync(ct)
 
         let events = ResizeArray()
```

Comment:
if doing overkill, wonder does reader offer length internally before you Read to the end so that could be passed as the Capacity (or maybe use batchSize so ResizeArray would never need to realloc)

Or you could skip worrying about it and just `return [| while reader.Read() -> parseRow reader |]`

Reply:
> or maybe use batchSize so ResizeArray would never need to realloc

I think the general case is < 10 events in the stream so I think allocating a whole batch might be counterproductive allocation wise 🤔

Reply:
I think I'll keep as a resize array for now. I'm not sure what the implications are of a while loop array expression 😅